### PR TITLE
proposal for ISC license

### PR DESCRIPTION
**Pull request by polettix**
Hi,
this is a proposal for the ISC license (see https://rt.cpan.org/Ticket/Display.html?id=66180)

Cheers,

```
Flavio.
```

### CODE REVIEW

**Comment by rjbs** [comment]
I vote :+1:

**Comment by karenetheridge** [comment]
I vote yes -- more licence documents are fine if they have any chance of being useful.

**Comment by Leont** [comment]
I vote yes on the idea, but the implementation is wrong. In particular the section
~~~
 +Copyright (c) 2004-2011 by Internet Systems Consortium, Inc. ("ISC")
+Copyright (c) 1995-2003 by Internet Software Consortium 
~~~
Should be something like
~~~
Copyright (c) {{$self->year}} by {{$self->_dotless_holder}}.
~~~

**Comment by polettix** [comment]
@Leont that seems to be the copyright statement for the licence itself, and as such it is not subject to change. The two lines are due to the acronym expansion change in 2003-2004 (ISC got changed from "Internet Software Consortium" to "Internet Systems Consortium, Inc.")

**Comment by Leont** [comment]
[Wikipedia](https://en.wikipedia.org/wiki/ISC_license) and the [open source initiative](https://opensource.org/licenses/ISC) disagree with you.

**Comment by polettix** [comment]
Makes sense actually. The OSI text, which is linked by the ISC site too so, to some extent, it is "blessed" by them. Is it better to the branch this PR is based on, or to rebase first and force-push the change in my repo?

**Comment by Leont** [comment]
rebase and force-push is fine. Do note "ISC" in the main text should also be replaced with "THE AUTHOR"

File: lib/Software/License/ISC.pm [hunk 1]
```diff
@@ -14,8 +14,7 @@ __DATA__
 __LICENSE__
 The ISC License
 
-Copyright (c) 2004-2011 by Internet Systems Consortium, Inc. ("ISC")
-Copyright (c) 1995-2003 by Internet Software Consortium
+Copyright (c) {{$self->year}} by {{$self->holder}}
```
Comment (polettix) [inline lib/Software/License/ISC.pm]: I saw that some other license files and Leon's suggestion used `_dotless_holder`. Is this really what we are supposed to do? Why not promote that method to a public one if it's the best practice?

**Comment by polettix** [comment]
Rebased and force-pushed.

Using public method `holder` instead of `_dotless_holder` as you suggested in the first place, because there's no dot at the end of the copyright line (as per official template) so it's up to the user to put that or not. Incidentally, I'd suggest to promote `_dotless_holder` to a public method, as to avoid confusion (am I *really* supposed to be able to use it?).

I'm not sure I'm understanding your comment about *"ISC" in the main text*, there is now no reference to it apart in the title (which I guess it's OK looking at the other licences).

**Comment by Leont** [comment]
> I'm not sure I'm understanding your comment about "ISC" in the main text, there is now no reference to it apart in the title (which I guess it's OK looking at the other licences).

Look again:
~~~~
THE SOFTWARE IS PROVIDED "AS IS" AND ISC DISCLAIMS ALL WARRANTIES
...
IN NO EVENT SHALL ISC BE LIABLE FOR
~~~~

**Comment by polettix** [comment]
This will teach me to look at the text in the repository, instead of dumbly using search in the browser inside the OSI page. The dark side of lazyness.

Considering that the search was successful (in not finding "ISC"), I opted for copy-pasting the text from the OSI website :-)

**Comment by karenetheridge** [comment]
> Incidentally, I'd suggest to promote _dotless_holder to a public method, as to avoid confusion (am I really supposed to be able to use it?).

+1

**Comment by Leont** [comment]
@rjbs this looks mergeable to me

**Comment by djzort** [comment]
11 years later, 6 years since "looks mergeable to me" Can we merge this?

**Comment by Leont** [comment]
We waited so long it needs an update for spdx support, but perhaps we should fix that after merging this.

**Comment by djzort** [comment]
Lets do it!

**Comment by Leont** [comment]
It turns out this doesn't index because in the mean time [such a module](https://metacpan.org/pod/Software::License::ISC) has arisen on CPAN. I guess I should ask for the permissions for that namespace

**Pull request merged**